src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1998,13 +2021,22 @@ private void HandleVariableDeclaration(IVariableDeclarationOperation operation)
                     }
                 }
 
+                // If we have an afterInitialization, then we must have an initializer to ensure we don't create empty regions that can't be cleaned up.
+                Debug.Assert(afterInitialization == null || initializer != null);
```

Comment:
Perhaps add `Debug.Assert((afterInitialization != null) == localSymbol.IsStatic);` since it looks like we're ignoring `afterInitialization` when `localSymbol` is not a static.

Reply:
Can't be quite this, as you can have a null afterInitialization if there's no initializer for the static variable. I'll add something similar.

---
In reply to: [175163581](https://github.com/dotnet/roslyn/pull/25458#discussion_r175163581) [](ancestors = 175163581)